packages/flutter/lib/src/material/snack_bar.dart

```diff
@@ -91,33 +100,24 @@ class _SnackBar extends StatelessComponent {
       )
     );
   }
-}
 
-class _SnackBarRoute extends TransitionRoute {
-  _SnackBarRoute({ Completer completer }) : super(completer: completer);
+  // API for Scaffold.addSnackBar():
 
-  bool get opaque => false;
-  Duration get transitionDuration => const Duration(milliseconds: 200);
-}
-
-Future showSnackBar({ BuildContext context, GlobalKey<PlaceholderState> placeholderKey, Widget content, List<SnackBarAction> actions }) {
-  final Completer completer = new Completer();
-  _SnackBarRoute route = new _SnackBarRoute(completer: completer);
-  _SnackBar snackBar = new _SnackBar(
-    route: route,
-    content: content,
-    actions: actions
-  );
-
-  // TODO(hansmuller): https://github.com/flutter/flutter/issues/374
-  assert(placeholderKey.currentState.child == null);
+  static Performance createPerformance(PerformanceStatusListener statusListener) {
+    assert(statusListener != null);
+    return new Performance(
+      duration: kSnackBarTransitionDuration,
+      debugLabel: 'SnackBar'
+    )..addStatusListener(statusListener);
```

Comment:
Why does this add the status listener for you?  Seems cleaner to just return the performance and have the caller add the status listener.